asktext.py

```diff
@@ -231,32 +258,60 @@ def ask(self, callback, preview_callback=None):
             self._preamble.insert(Tk.END, self.preamble_file)
             box.pack(fill="x", pady=5, expand=True)
 
+            # Frame box for tex command
+            tex_command_tk_str = Tk.StringVar()
+            tex_command_tk_str.set(self.current_texcmd)
+
+            box = Tk.Frame(self._frame, relief="groove", borderwidth=2)
+            label = Tk.Label(box, text="TeX command:")
+            label.pack(pady=2, padx=5, anchor="w")
+            for tex_command in self.TEX_COMMANDS:
+                Tk.Radiobutton(box, text=tex_command, variable=tex_command_tk_str,
+                               value=tex_command).pack(side="left", expand=False, anchor="w")
+            box.pack(fill="x", pady=5, expand=True)
+
             # Frame box for scale factor and reset buttons
             box = Tk.Frame(self._frame, relief="groove", borderwidth=2)
             label = Tk.Label(box, text="Scale factor:")
             label.pack(pady=2, padx=5, anchor="w")
-            explanation = """Use RESET to set scale factor to the value this node has been
-created with."""
-            label = Tk.Label(box, justify="left", text=explanation)
-            label.pack(pady=2, padx=5, anchor="w")
-            explanation = """Use GLOBAL to set scale factor to the value of the previously
-edited node in Inkscape."""
-            label = Tk.Label(box, justify="left", text=explanation)
-            label.pack(pady=2, padx=5, anchor="w")
-            self._scale = Tk.Scale(box, orient="horizontal", from_=0.1, to=10, resolution=0.1)
+
+            validation_command = (root.register(self.validate_spinbox_input),
+                                  '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+            self._scale = Tk.Spinbox(box, from_=0.001, to=10, increment=0.001, validate="key",
+                                     validatecommand=validation_command)
             self._scale.pack(expand=True, fill="x", ipady=4, pady=5, padx=5, side="left", anchor="e")
-            self._scale.set(self.scale_factor_after_loading())
+            self._scale.delete(0, "end")
+            self._scale.insert(0, self.scale_factor_after_loading())
 
             reset_scale = self.current_scale_factor if self.current_scale_factor else self.global_scale_factor
-            self._reset_button = Tk.Button(box, text="Reset ({0:.1f})".format(reset_scale),
+            self._reset_button = Tk.Button(box, text="Reset ({0:.3f})".format(reset_scale),
                                            command=self.reset_scale_factor)
             self._reset_button.pack(ipadx=10, ipady=4, pady=5, padx=5, side="left")
-            self._global_button = Tk.Button(box, text="Global ({0:.1f})".format(self.global_scale_factor),
+            self._global_button = Tk.Button(box, text="From previous node ({0:.3f})".format(self.global_scale_factor),
```

Comment:
I think "As previous" sounds better than "From previous"

Reply:
Labels in GTK and TK versions must match

Reply:
Agreed